drivers/flash/flash_stm32_generic.c

```diff
@@ -0,0 +1,243 @@
+/*
+ * Copyright (c) 2017 BayLibre, SAS
+ * Copyright (c) 2019 Linaro Limited
+ * Copyright (c) 2020 Andreas Sandberg
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(flash_stm32generic, CONFIG_FLASH_LOG_LEVEL);
+
+#include <kernel.h>
+#include <device.h>
+#include <string.h>
+#include <drivers/flash.h>
+#include <init.h>
+#include <soc.h>
+
+#include "flash_stm32.h"
+
+#define WRITE_BLOCK_SIZE DT_PROP(DT_INST(0, soc_nv_flash), write_block_size)
```

Comment:
With the introduction of https://github.com/zephyrproject-rtos/zephyr/pull/25947 the WRITE_BLOCK_SIZE should be also assigned to `flash_parameters.write_block_size` , specific for the flash device.

Reply:
Done.